File: components/cmd/pkg/utilities/logger.go
// Copyright (c) 2020 Xiaozhe Yao & AICAMP.CO.,LTD
//
// This software is released under the MIT License.
// https://opensource.org/licenses/MIT

package utilities

import (
	"github.com/sirupsen/logrus"
	"path/filepath"
	"time"
)

// DefaultLogger is the instance shared by all modules
var DefaultLogger *logrus.Logger

// NewDefaultLogger returns the Logger Object
func NewDefaultLogger() *logrus.Logger {
	if DefaultLogger != nil {
		return DefaultLogger
	}
	logPath := filepath.Join(GetBasePath(), "logs", "system")
	return NewLogger(logPath)
}

// NewLogger returns a new Logger
func NewLogger(logPath string) *logrus.Logger {
	var logger *logrus.Logger
	CreateFolderIfNotExist(filepath.Dir(logPath))
	rotateFileHook, err := NewRotateFileHook(RotateFileConfig{
		Filename:   logPath,
		MaxSize:    50, // megabytes
		MaxBackups: 3,
		MaxAge:     28, //days
		Level:      logrus.InfoLevel,
		Formatter: &logrus.JSONFormatter{
			TimestampFormat: time.RFC822,
		},
	})
	if err != nil {
		logrus.Error("Cannot Initialize Logger..")
		logrus.Error(err.Error())
	}
	logger = logrus.New()
	logger.AddHook(rotateFileHook)
	return logger
}

File: components/aipm/logger.py
# Copyright (c) 2020 Xiaozhe Yao & AICAMP.CO.,LTD & AutoAI.org
# 
# This software is released under the MIT License.
# https://opensource.org/licenses/MIT
import os
import json

class Logger(object):
    def __init__(self, target):
        self.target = target
    
    def write(self, jsonObject):
        with open(self.target, 'w+') as f:
            json.dump([jsonObject], f)
    
    def append(self, jsonObject):
        if os.stat(self.target).st_size == 0:
            self.write(jsonObject)
        else:
            with open(self.target, 'w+') as f:
                data = json.load(f)
                data.append(jsonObject)
                json.dump(data, f)